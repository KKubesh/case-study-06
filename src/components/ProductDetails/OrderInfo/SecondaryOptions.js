import React, {Component} from 'react';

class SecondaryOptions extends Component {
    render () {
        return (
            <div className="row-flex addto-margin" data-testid="secondary-options">
                <button className="button-grey" data-testid="add-to-registry">Add to registry</button>
                <button className="button-grey" data-testid="add-to-list">Add to list</button>
                <button className="button-grey" data-testid="share">share</button>
            </div>
        )
    }
}

export default SecondaryOptions;