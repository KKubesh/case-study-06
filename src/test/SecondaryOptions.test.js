import React from 'react';
import { render, cleanup } from 'react-testing-library';
import SecondaryOptions from '../components/ProductDetails/OrderInfo/SecondaryOptions';

afterEach(cleanup);

test('<SecondaryOptions />', () => {
    const { getByTestId, container } = render(<SecondaryOptions/>);
    expect(getByTestId('add-to-registry').textContent).toBe("Add to registry");
    expect(getByTestId('add-to-list').textContent).toBe("Add to list");
    expect(getByTestId('share').textContent).toBe("share");
    expect(container.firstChild).toMatchSnapshot();
});