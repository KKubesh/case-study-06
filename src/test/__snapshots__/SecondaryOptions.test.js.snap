// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SecondaryOptions /> 1`] = `
<div
  class="row-flex addto-margin"
  data-testid="secondary-options"
>
  <button
    class="button-grey"
    data-testid="add-to-registry"
  >
    Add to registry
  </button>
  <button
    class="button-grey"
    data-testid="add-to-list"
  >
    Add to list
  </button>
  <button
    class="button-grey"
    data-testid="share"
  >
    share
  </button>
</div>
`;
